src/kernel/mempool_entry.h

```diff
@@ -152,13 +154,13 @@ class CTxMemPoolEntry
     }
 
     uint64_t GetCountWithDescendants() const { return nCountWithDescendants; }
-    uint64_t GetSizeWithDescendants() const { return nSizeWithDescendants; }
+    int64_t GetSizeWithDescendants() const { return nSizeWithDescendants; }
```

Comment:
Re-reviewing this basically from scratch, I stumble here. `int32_t` should be big enough to represent the weight of more than 5,000 max standard weight transactions. Are we actually bumping into overflow issues with `int32_t` somewhere for size with descendants?

I see that this was brought up before, so I assume it was left a 64-bit value to limit the scope of this PR. Is that right?

Reply:
> Are we actually bumping into overflow issues with `int32_t` somewhere for size with descendants?

I don't think so.

However, UBSan raises "signed-integer-overflow" warnings about `nSizeWithAncestors += modifySize;` etc.